js/engine/Main.ml

```diff
@@ -25,11 +25,24 @@ let _ =
        method deleteFile filename = Sys.remove (Js.to_string filename)
 
        method setParsePattern (func : bool -> Lang.t -> string -> 'a) =
-         Parse_pattern.parse_pattern_ref := func
+         Parse_pattern.parse_pattern_ref :=
+           fun print_errors lang str ->
+             match lang with
+             | Lang.Yaml -> Yaml_to_generic.any str
```

Comment:
maybe add a comment like
(* The Yaml parser is already in engine.js because it is a core component needed to parse the rules.
 * No need to call the external parser.js *)

Reply:
BTW the JS/JSON/TS parsers are currently also in engine.js. See src/parsing/Parse_target.ml
for similar reasons to the Yaml parser; because rules can be written in JSON, and
to parse JSON we abuse the JS/TS parser.
Maybe add also a comment like (* coupling: See src/Parsing/Parse_target.ml list of languages builtin engine.js *)

Reply:
Will do 👍